### PR TITLE
ProxyBay related cleanups

### PR DESCRIPTION
**Pull request by funkydude**
Removes an old, dead ruleset.
Splits one out.
Updates current one with some comments.

More work needed in a separate PR.

### CODE REVIEW

File: src/chrome/content/rules/ProxyBay.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<!--
+	HSTS: max-age=15552000; includeSubDomains; preload || Not in the list yet
```
Comment (J0WI) [inline src/chrome/content/rules/ProxyBay.xml]: It is: https://dxr.mozilla.org/mozilla-central/source/security/manager/ssl/nsSTSPreloadList.inc#5074

File: src/chrome/content/rules/ThePirateBay.xml [hunk 2]
```diff
@@ -11,15 +11,14 @@
 	<target host="www.thepiratebay.org" />
 	<target host="pirates-forum.org" />
 	<target host="www.pirates-forum.org" />
-	<target host="kanyebay.com" />
-	<target host="www.kanyebay.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/ThePirateBay.xml]: host is still up

**Comment by funkydude** [comment]
@J0WI Resolved

**Comment by J0WI** [comment]
preloaded sites don't need to be added to HTTPS Everywhere. You can safely remove the proxybay.la rules.

**Comment by funkydude** [comment]
@J0WI there doesn't appear to be a clear consensus on that right now: https://github.com/EFForg/https-everywhere/issues/4396

On top of that, since this domain will likely change again soon, I do not wish to delete and re-create it for whatever time it takes for the new domain to be added to the list. It will also need forwarding from the old to the new domain. So, I'm leaving it in for this one in specific.

**Comment by J0WI** [comment]
I think this will be fixed in #4462

**Comment by jeremyn** [comment]
@funkydude @J0WI I noticed this because it's old and `top-1k`. What's blocking this?

**Comment by J0WI** [comment]
@jeremyn the point is, that sites in the [HSTS preload](https://hstspreload.appspot.com/) list don't need to be in HTTPS Everywhere, because they are already forced by the browser. I think browsers doings this well than HTTPS Everywhere and it would shrink our extension.

Quoting @jsha from the mailing list:
> It's fine to remove an auto-generated HSTS rule, if:
> - Its hosts are now fully covered in the HSTS preload list.
> - The secure cookie rules are not necessary (e.g. the site secures all its cookies, *or* only sets cookies that are scoped exactly to the covered HSTS domain).

But we haven't cleaned up our rules yet (see https://github.com/EFForg/https-everywhere/issues/4396#issuecomment-240007647).

**Comment by jeremyn** [comment]
@J0WI So does that mean the blocker is you want @funkydude to remove `ProxyBay.xml` from the pull request?

**Comment by J0WI** [comment]
Either this or at least adding a comment, that we can easily clean up preloaded hosts if we want to do this.

**Comment by J0WI** [comment]
I think I'll prefer #5013 which is more up to date and also includes more PirateBay related hosts.

**Comment by J0WI** [comment]
closed by #5013

**Pull request closed without merging**